File: app/src/main/java/com/mcafi/calcetto/SettingsActivity.java
package com.mcafi.calcetto;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.FirebaseFirestoreSettings;
import com.mcafi.calcetto.model.User;

import java.util.HashMap;
import java.util.Map;

public class SettingsActivity extends AppCompatActivity implements View.OnClickListener {

    private FirebaseAuth mAuthReg = FirebaseAuth.getInstance();
    private FirebaseFirestore db = initializeDatabase();
    private FirebaseUser firebaseUser = mAuthReg.getCurrentUser();
    private final DocumentReference userRef = db.collection("utenti").document(firebaseUser.getUid());
    private User user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        Toolbar toolbar = (Toolbar) findViewById(R.id.settingsToolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setDisplayShowTitleEnabled(false);

        final TextView nameText = findViewById(R.id.nameText);
        final TextView usernameText = findViewById(R.id.usernameText);
        TextView saveSettings = findViewById(R.id.saveSettings);
        saveSettings.setOnClickListener(this);

        userRef.get().addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
            @Override
            public void onSuccess(DocumentSnapshot documentSnapshot) {
                User user = documentSnapshot.toObject(User.class);
                nameText.setText(user.getName());
                usernameText.setText(user.getUsername());
            }
        });
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    @Override
    public void onClick(View v) {
        TextView nameText = findViewById(R.id.nameText);
        TextView usernameText = findViewById(R.id.usernameText);
        Map<String, Object> user = new HashMap<>();
        user.put("name", nameText.getText().toString());
        user.put("username", usernameText.getText().toString());
        userRef.update(user);
        Toast.makeText(getApplicationContext(), "Modifiche salvate!", Toast.LENGTH_LONG);
    }

    private FirebaseFirestore initializeDatabase() {
        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings.Builder()
                .setPersistenceEnabled(true)
                .build();
        FirebaseFirestore db = FirebaseFirestore.getInstance();
        db.setFirestoreSettings(settings);
        return db;
    }
}

File: app/src/main/java/com/mcafi/calcetto/SettingsActivity.kt
package com.mcafi.calcetto

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.FirebaseFirestoreSettings
import com.mcafi.calcetto.db.DbCreator
import com.mcafi.calcetto.model.User
import kotlinx.android.synthetic.main.activity_match_view.*
import kotlinx.android.synthetic.main.activity_settings.*


class SettingsActivity : AppCompatActivity(), View.OnClickListener {
    private val mAuthReg = FirebaseAuth.getInstance()
    private val db = initializeDatabase()
    private val firebaseUser = mAuthReg.currentUser
    private val userRef = db.collection("utenti").document(firebaseUser!!.uid)
    private lateinit var user: User
    private val DbSql = DbCreator(this)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        setSupportActionBar(settings_toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.setDisplayShowTitleEnabled(false)
        val nameText = findViewById<TextView>(R.id.et_settings_name)
        val usernameText = findViewById<TextView>(R.id.et_settings_username)
        val saveSettings = findViewById<TextView>(R.id.btn_settings_save)
        saveSettings.setOnClickListener(this)


        userRef.get().addOnSuccessListener { documentSnapshot ->
            user = documentSnapshot.toObject(User::class.java)!!
            nameText.text = user.name
            usernameText.text = user.username
        }

        val notify_user=DbSql.getUser(mAuthReg.currentUser!!.uid)
        if(notify_user==1){
            notification_switch.isChecked=true
        }
        else if(notify_user==0){
            notification_switch.isChecked=false
        }
        notification_switch.setOnClickListener(this)
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.btn_settings_save -> {
                user.name = et_settings_name.text.toString()
                user.username = et_settings_username.text.toString()
                userRef.update(user.toMap())
                Toast.makeText(applicationContext, "Modifiche salvate!", Toast.LENGTH_LONG).show()
                startActivity(Intent(this, MainActivity::class.java).putExtra("TAB", 2))
            }
            R.id.logoutButton -> {
                FirebaseAuth.getInstance().signOut()
                startActivity(Intent(this, LoginActivity::class.java))
            }
            R.id.notification_switch->{
                if(notification_switch.isChecked){
                    DbSql.execQuery("UPDATE user SET notify = 1 WHERE id_user= '"+mAuthReg.currentUser!!.uid+"';");
                    Toast.makeText(applicationContext, "Uno", Toast.LENGTH_LONG).show()
                }
                else{
                    DbSql.execQuery("UPDATE user SET notify = 0 WHERE id_user= '"+mAuthReg.currentUser!!.uid+"'  ;");
                    Toast.makeText(applicationContext, "Zero", Toast.LENGTH_LONG).show()
                }
            }

        }
    }

    private fun initializeDatabase(): FirebaseFirestore {
        val settings = FirebaseFirestoreSettings.Builder()
                .setPersistenceEnabled(true)
                .build()
        val db = FirebaseFirestore.getInstance()
        db.firestoreSettings = settings
        return db
    }
}